propagate transitions to d3fc-data-join

ColinEberhardt opened the pull request.

chrisprice commented on examples/chart.js:
```diff
@@ -33,6 +33,4 @@ function render() {
         renderComponent();
     }
-
-    requestAnimationFrame(render);
```
Does this not break the crosshair example?

chrisprice commented on package.json:
```diff
@@ -19,17 +19,17 @@
   "devDependencies": {
-    "d3-axis": "^1.0.3",
-    "d3-shape": "^1.0.3",
+    "babel-polyfill": "^6.16.0",
```
Would it make sense to put this in d3fc-scripts? (Maybe just open an issue for now)

chrisprice commented on src/svg/gridline.js:
```diff
@@ -19,6 +19,11 @@ export default () => {
 
+        if (selection.selection) {
+            xJoin.transition(selection);
+            yJoin.transition(selection);
```
Did you decide against transitioning the containers?

ColinEberhardt commented:
@chrisprice I've updated the code, addressing those points.

The gridlines component really needs some logic similar to axis in order to make transitions work well. I think I'll raise an issue for that and address after d3fc v.next release. I'm happy that d3fc-series now supports transitions, which is the main one to solve.

chrisprice commented:
Agreed, I think they probably all need it.
